news/7737.feature

```diff
@@ -0,0 +1 @@
+Added ``__repr__`` for ``Configuration`` to make debugging a bit easier.
```

Comment:
I would put this in a file with a `.trivial` ending, since it's not user-facing.